qiskit/transpiler/passes/synthesis/plugin.py

```diff
@@ -243,6 +246,33 @@ def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **
 :func:`qiskit.transpiler.passes.synthesis.plugin.unitary_synthesis_plugin_names`
 function.
 
+High-level Synthesis Plugins
+----------------------------
+
+To use a high-level synthesis plugin, you first instantiate an :class:`.HLSConfig` to
+store the names of the plugins to use for various high-level objects.
+For example::
+
+    HLSConfig(permutation=["acg"], clifford=["layers"], linear_function=["pmh"])
```

Comment:
Can you comment on the possible keyword arguments here? e.g. how does the user know that `permutation`, `clifford`, and `linear_function` are possibilities?

Reply:
I've pushed up a8244435e2 to expand on this - is it understandable for you?